### PR TITLE
loop: versioned MuSig2 HTLCs

### PR DESCRIPTION
**Pull request by bhandras**
This pull requests includes a draft version of changes required to make Taproot HTLCs optionally MuSig2 1.0 compatible.

Depends on: 
~- https://github.com/lightningnetwork/lnd/pull/7171~
~- https://github.com/lightninglabs/lndclient/pull/141~

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: swap/htlc.go [hunk 1]
```diff
@@ -122,8 +122,8 @@ var (
 	// QuoteHtlcP2TR is a template script just used for sweep fee
 	// estimation.
 	QuoteHtlcP2TR, _ = NewHtlcV3(
-		^int32(0), dummyPubKey, dummyPubKey, dummyPubKey, dummyPubKey,
-		quoteHash, &chaincfg.MainNetParams,
+		input.MuSig2Version100RC2, ^int32(0), dummyPubKey, dummyPubKey,
```
Comment (guggero) [inline swap/htlc.go:125]: This uses 1.0 while the other changes in the PR use 0.4. Is that on purpose?

File: swap/htlc.go [hunk 2]
```diff
@@ -560,26 +560,25 @@ func newHTLCScriptV3(cltvExpiry int32, senderInternalKey, receiverInternalKey,
 		return nil, err
 	}
 
+	var aggregateKey *musig2.AggregateKey
 	// Calculate the internal aggregate key.
-	aggregateKey, _, _, err := musig2.AggregateKeys(
+	aggregateKey, err = input.MuSig2CombineKeys(muSig2Version,
```
Comment (guggero) [inline swap/htlc.go:565]: nit: formatting.

**Comment by bhandras** [comment]
> Changes so far look good to me. So there isn't going to be a new swap version that can be used to decide between MuSig2 0.4 and 1.0? Or is this just not implemented yet since it's a draft?

Protocol version HTLC_V3 did include experimental MuSig2 support for loop out and we'll keep those users on 0.4 (see other PRs). For new clients we'll require everyone to use 1.0.

File: swap/htlc.go (hunk 1 above)
Comment (bhandras) [reply swap/htlc.go:125]: Yeah this PR is just making it possible to pass in the version if needed. We just need this specific dummy htlc for fee estimation so the version can be anything here.

**Review by sputn1ck — APPROVED** [review]
LGTM!

File: loopd/run.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ var (
 	LoopMinRequiredLndVersion = &verrpc.Version{
 		AppMajor: 0,
 		AppMinor: 15,
-		AppPatch: 4,
+		AppPatch: 99,
```
Comment (sputn1ck) [inline loopd/run.go:30]: That will be changed to 0.16 right?

File: loopin.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/lightninglabs/loop/loopdb"
 	"github.com/lightninglabs/loop/swap"
 	"github.com/lightningnetwork/lnd/chainntnfs"
-	"github.com/lightningnetwork/lnd/channeldb"
+	invpkg "github.com/lightningnetwork/lnd/invoices"
```
Comment (GeorgeTsagk) [inline loopin.go:21]: was there a conflict with `invoices`?

**Review by GeorgeTsagk — APPROVED** [review]
LGTM :loop:

File: loopin.go (hunk 4 above)
Comment (bhandras) [reply loopin.go:21]: It's just a convention that we seemed to have picked up in the LND codebase so I was just meant to conform to that.

**Pull request merged**